### ISSUE TITLE
[Bugs] MySQL "TYPE=MyIsam" option still remains

### ISSUE THREAD

**Issue by mystralkk** [Geeklog-Core/geeklog#548]
**Reported by mystralkk on 26 Oct 2011 12:35**

**Version:** 1.8.1

[Mantis Bugtracker #1403](http://project.geeklog.net/tracking/view.php?id=1403)

**Comment by mystralkk** [comment]
**Comment posted by dhaun on 1 May 2012 13:43**

Just to clarify: The issue is about the use of the TYPE keyword, which is no longer supported in MySQL 5.x (5.4.x or 5.5? I found conflicting information).

As of MySQL 5.5, InnoDB is the default type (instead of MyISAM), but that is another issue and does not cause problems in Geeklog, since we always specify the table type.

The ENGINE keyword was introduced in MySQL 4.0.18 and 4.1.2 while the minimum requirement for Geeklog is now MySQL 4.1.3.

In other words, it should be okay to replace TYPE with ENGINE, even in those old files.

Alternatively, we could silently replace TYPE with ENGINE in dbQuery() in the MySQL and MySQLi bindings. That would then also take care of unsupported plugins.

**Comment by mystralkk** [comment]
**Comment posted by mystralkk on 30 May 2012 09:52**

As of Geeklog-2.0beta, two files still include &quot;TYPE=MyIsam&quot; phrase:

plugins/links/sql/mysql_updates.php at line 59
plugins/polls/sql/mysql_updates.php at line 54
